servers/zts/src/main/java/com/yahoo/athenz/zts/utils/ZTSUtils.java

```diff
@@ -305,7 +311,53 @@ public static Identity generateIdentity(InstanceCertManager certManager, final S
     }
 
     public static SSLContext createServerClientSSLContext(PrivateKeyStore privateKeyStore) {
+        if (Boolean.parseBoolean(System.getProperty(ATHENZ_PROP_USE_LOCAL_CERT_AND_KEY_FOR_SSL, ATHENZ_DEFAULT_USE_LOCAL_CERT_AND_KEY_FOR_SSL))) {
+            return getSslContextAthenzCert(privateKeyStore);
+        } else {
+            return getSslContextExternalCert(privateKeyStore);
+        }
+    }
+
+    private static SSLContext getSslContextAthenzCert(PrivateKeyStore privateKeyStore) {
+        final String trustStorePath = System.getProperty(ATHENZ_PROP_TRUSTSTORE_PATH);
+        if (trustStorePath == null) {
+            LOGGER.error("Unable to create client ssl context: no truststore path specified");
+            return null;
+        }
+        final String certPath = System.getProperty(ATHENZ_PROP_LOCAL_CERT_PATH_FOR_SSL);
+        if (certPath == null) {
+            LOGGER.error("Unable to create client ssl context: no local ssl cert path specified");
+            return null;
+        }
+        if (!new File(certPath).exists()) {
+            LOGGER.error("Unable to create client ssl context: local ssl cert not found in {}", certPath);
+            return null;
+        }
+        final String keyPath = System.getProperty(ATHENZ_PROP_LOCAL_KEY_PATH_FOR_SSL);
+        if (keyPath == null) {
+            LOGGER.error("Unable to create client ssl context: no local ssl key path specified");
+            return null;
+        }
+        if (!new File(keyPath).exists()) {
+            LOGGER.error("Unable to create client ssl context: local ssl key not found in {}", keyPath);
+            return null;
+        }
+
+        final String trustStorePassword = System.getProperty(ATHENZ_PROP_TRUSTSTORE_PASSWORD);
+        final String trustStorePasswordAppName = System.getProperty(ATHENZ_PROP_TRUSTSTORE_PASSWORD_APPNAME);
+        final String password = getApplicationSecret(privateKeyStore, trustStorePasswordAppName, trustStorePassword);
+        char[] passwordChars = getPasswordChars(password);
+        try {
+            KeyRefresher keyRefresher = Utils.generateKeyRefresher(trustStorePath, passwordChars, certPath, keyPath);
+            keyRefresher.startup();
+            return Utils.buildSSLContext(keyRefresher.getKeyManagerProxy(), keyRefresher.getTrustManagerProxy());
+        } catch (Exception ex) {
+            LOGGER.error("Unable to create client ssl context. Error: {}", ex.getMessage());
+            return null;
+        }
+    }
 
+    private static SSLContext getSslContextExternalCert(PrivateKeyStore privateKeyStore) {
```

Comment:
this should be called getSslContextUsingKeyStores

Reply:
Fixed